cw-orch/src/daemon/queriers/node.rs

```diff
@@ -182,4 +182,39 @@ impl Node {
         // return error if tx not found by now
         Err(DaemonError::TXNotFound(hash, MAX_TX_QUERY_RETRIES))
     }
+
+    /// Find TX by events
+    pub async fn find_tx_by_events(&self, events: Vec<String>, page: Option<PageRequest>, order_by: Option<OrderBy>) -> Result<Vec<CosmTxResponse>, DaemonError> {
+        let mut client =
+            cosmos_modules::tx::service_client::ServiceClient::new(self.channel.clone());
+
+        let request = cosmos_modules::tx::GetTxsEventRequest{ 
+            events: events.clone(),
+            pagination: page,
+            order_by: order_by.unwrap_or(OrderBy::Desc).into()
+        };
+
+        for _ in 0..MAX_TX_QUERY_RETRIES {
+            match client.get_txs_event(request.clone()).await {
+                Ok(tx) => {
+                    let resp = tx.into_inner().tx_responses;
+                    if resp.is_empty() {
+                        log::debug!("Not TX by events found");
+                        log::debug!("Waiting 10s");
+                        sleep(Duration::from_secs(10)).await;
```

Comment:
Can this use the dynamic wait like the tx_by_hash one?

Reply:
Resolved in https://github.com/AbstractSDK/cw-orchestrator/pull/57/commits/a1fa7fc5b642daa110767186f3ebbe4adfdaa2c9